packages/browser/src/core/page/add-page-context.ts

```diff
@@ -0,0 +1,174 @@
+import { Campaign } from '@segment/analytics-core'
+import { pick } from '../../lib/pick'
+import { EventProperties, SegmentEvent } from '../events'
+import { getDefaultPageContext, PageContext } from './get-page-context'
+import { gracefulDecodeURIComponent } from '../query-string/gracefulDecodeURIComponent'
+import { tld } from '../user/tld'
+import { getVersionType } from '../../lib/version-type'
+import { UniversalStorage, CookieStorage } from '../storage'
+import jar from 'js-cookie'
+import { version } from '../../generated/version'
+
+let cookieOptions: jar.CookieAttributes | undefined
+function getCookieOptions(): jar.CookieAttributes {
+  if (cookieOptions) {
+    return cookieOptions
+  }
+
+  const domain = tld(window.location.href)
+  cookieOptions = {
+    expires: 31536000000, // 1 year
+    secure: false,
+    path: '/',
+  }
+  if (domain) {
+    cookieOptions.domain = domain
+  }
+
+  return cookieOptions
+}
+
+type Ad = { id: string; type: string }
+
+/**
+ *
+ * @param obj e.g. { foo: 'b', bar: 'd'}
```

Comment:
This adds a "feature" / fixes a bug where now we will calculate context.campaign if a user sets context.page.search to an object (as people sometimes do).